lib/pages/chat_list/client_chooser_button.dart

```diff
@@ -89,66 +82,93 @@ class ClientChooserButton extends StatelessWidget {
+                    matrix.accountBundles[bundle]!.single!.userID != bundle)
+                  PopupMenuItem(
+                    value: null,
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      mainAxisSize: MainAxisSize.min,
+                      children: [
+                        Text(
+                          bundle!,
+                          style: TextStyle(
+                            color:
+                                Theme.of(context).textTheme.titleMedium!.color,
+                            fontSize: 14,
+                          ),
+                        ),
+                        const Divider(height: 1),
+                      ],
                     ),
                   ),
-                  const SizedBox(width: 12),
-                  IconButton(
-                    icon: const Icon(Icons.edit_outlined),
-                    onPressed: () => controller.editBundlesForAccount(
-                      client.userID,
-                      bundle,
+                ...matrix.accountBundles[bundle]!.map(
+                  (client) => PopupMenuItem(
+                    value: client,
+                    child: FutureBuilder<Profile?>(
+                      // analyzer does not understand this type cast for error
+                      // handling
+                      //
+                      // ignore: unnecessary_cast
+                      future: (client!.fetchOwnProfile() as Future<Profile?>)
+                          .onError((e, s) => null),
+                      builder: (context, snapshot) => Row(
+                        children: [
+                          Avatar(
+                            mxContent: snapshot.data?.avatarUrl,
+                            name: snapshot.data?.displayName ??
+                                client.userID!.localpart,
+                            size: 32,
+                            fontSize: 12,
+                          ),
+                          const SizedBox(width: 12),
+                          Expanded(
+                            child: Text(
+                              snapshot.data?.displayName ??
+                                  client.userID!.localpart!,
+                              overflow: TextOverflow.ellipsis,
+                            ),
+                          ),
+                          const SizedBox(width: 12),
+                          IconButton(
+                            icon: const Icon(Icons.edit_outlined),
+                            onPressed: () async {
+                              await controller.editBundlesForAccount(
+                                client.userID,
+                                bundle,
+                              );
+                              await client.oneShotSync();
+
+                              if (context.mounted) {
+                                setState(() {});
```

Comment:
This does only update the GUI if the change was done on this device. What if the change was triggered on a different device? We should use a StreamBuilder here and listen to the account data changes instead

Reply:
I'll get that written now